client/tx/tx.go

```diff
@@ -396,6 +399,54 @@ func Sign(txf Factory, name string, txBuilder client.TxBuilder, overwriteSig boo
 		return err
 	}
 
+	// Modify Ledger signature to use EIP712 extension
+	info, err := txf.Keybase().Key(name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving key type: %v\n", err)
+	}
+
+	if info.GetType().String() == "ledger" {
+		// Get extension builder to set Web3 extension
+		extensionBuilder, ok := txBuilder.(authtx.ExtensionOptionsTxBuilder)
+		if !ok {
+			return fmt.Errorf("Error setting extension options: cannot cast to ExtensionOptionsTxBuilder")
+		}
+
+		// Parse Chain ID as UInt
+		chainID, err := etherminttypes.ParseChainID(txf.chainID)
+		if err != nil {
+			return fmt.Errorf("Error parsing chain id: %v\n", err)
+		}
+
+		// Add ExtensionOptionsWeb3Tx extension
+		var option *codectypes.Any
+		option, err = codectypes.NewAnyWithValue(&etherminttypes.ExtensionOptionsWeb3Tx{
+			FeePayer:         info.GetAddress().String(),
+			TypedDataChainID: chainID.Uint64(),
+			FeePayerSig:      sigBytes,
+		})
+		extensionBuilder.SetExtensionOptions(option)
+
+		// Set blank signature data with Amino Sign Type
+		// (Regardless of input signMode, Evmos requires an Amino payload for Ledger)
+		sigData = signing.SingleSignatureData{
+			SignMode:  signing.SignMode_SIGN_MODE_LEGACY_AMINO_JSON,
+			Signature: nil,
+		}
+		sig = signing.SignatureV2{
+			PubKey:   pubKey,
+			Data:     &sigData,
+			Sequence: txf.Sequence(),
+		}
+
+		err = txBuilder.SetSignatures(sig)
+		if err != nil {
+			return fmt.Errorf("Unable to set signatures on payload: %v\n", err)
+		}
+
+		return nil
+	}
```

Comment:
How can we abstract this functionality in a way that the SDK can support it natively? Ideas:

- Create a new function (or interface) field in the `client.Context` or the `Factory` struct and call this function